lib/core/src/display/shape/glyph/system.rs

```diff
@@ -92,8 +92,9 @@ impl Line {
     /// The replacing strings will reuse glyphs which increases performance of rendering text.
     pub fn replace_text<Chars>(&mut self, chars:Chars, fonts:&mut FontRegistry)
     where Chars : Iterator<Item=char> + Clone {
-        let font = fonts.get_render_info(self.font_id);
-        let pen  = PenIterator::new(self.baseline_start,self.height,chars.clone(),font);
+        let font        = fonts.get_render_info(self.font_id);
+        let chars_count = chars.clone().count();
```

Comment:
What is chars and how costly is the clone here?

Reply:
This is iterator over &str or Vec<char>. Cloning should be just a copying reference and length